core/string/ustring.cpp

```diff
@@ -3507,6 +3507,48 @@ char32_t String::unicode_at(int p_idx) const {
 	return operator[](p_idx);
 }
 
+String String::get_indent() const {
+	String indent;
+	bool has_text = false;
+	int line_start = 0;
+
+	for (int i = 0; i < length(); i++) {
+		const char32_t c = operator[](i);
+		if (c == '\n') {
+			line_start = i + 1;
+		} else if (c > 32) {
+			has_text = true;
+			indent = substr(line_start, i - line_start);
+			break; // Indentation of the first line that has text.
+		}
+	}
+	if (!has_text) {
+		return *this;
+	}
+	return indent;
+}
+
+String String::indent(const String &p_indent) const {
```

Comment:
For the record we seem to have
```
core/io/json.h
72:     static String _make_indent(const String &p_indent, int p_size);
```
which could maybe be modified to use this instead if we're adding it to `String`.

Reply:
This one is just making the correct level of indent string instead of indenting. Can be probably replaced by `p_indent.repeat(p_size)` :)